src/functions/sortGoods.js

```diff
@@ -0,0 +1,40 @@
+import { ALPHABET, NO_SORT, LENTGH } from '../constances';
+
+export function sortGoods(
+  sortBy,
+  setSortBy,
+  setReversed,
+  setGoods,
+  originalGoods,
+  reversed,
+) {
+  switch (sortBy) {
+    case NO_SORT:
+      setReversed(false);
+      setGoods(originalGoods);
+      setSortBy(NO_SORT);
+      break;
+    case ALPHABET: {
+      setSortBy(ALPHABET);
+      const sortedGoods = originalGoods.sort(
+        (good1, good2) => good1.localeCompare(good2),
+      );
+
+      setGoods(reversed ? sortedGoods.reverse() : sortedGoods);
+      break;
+    }
+
+    case LENTGH: {
+      setSortBy(LENTGH);
+      const sortedGoods = originalGoods.sort(
```

Comment:
The sort function modifies the original array. You should make a copy before sorting to prevent unwanted side effects.

Reply:
I understand but originalGoods !== goodsFromServer. It is originalGoods === [...goodsFromServer]
So I don't modify the goodsFromServer